src/map/status.c

```diff
@@ -1560,6 +1560,42 @@ static int status_fixed_revive(struct block_list *bl, unsigned int per_hp, unsig
 	return 1;
 }
 
+/**
+ * a sub-function from 'status_check_skilluse' function, which can be use for plugin manipulation
+ * check 'disabled_skills' flag settings from `db\(pre-)re\map_zone_db.conf`
+ * @param src        The source object can be BL_PC, BL_MOB, BL_PET, BL_HOM, BL_MER, BL_ELEM type
+ * @param st         The status_data of the source object, use to check MD_BOSS of BL_MOB type
+ * @param skill_id   The skill ID use
+ * @return false if the skill ID for the src is denied, true otherwise
+ **/
+static bool status_check_skilluse_mapzone(struct block_list *src, struct status_data *st, uint16 skill_id)
```

Comment:
Either use `int` as return type or change the function's name to a predicate.

https://github.com/HerculesWS/Hercules/wiki/Coding-Style#chapter-16-function-return-values-and-names

> If the name of a function is an action or an imperative command, the function should return an error-code integer. If the name is a predicate, the function should return a "succeeded" boolean.

Reply:
I cannot agree with this
the function name already said **CHECK**, means must return true on success

Reply:
> the function name already said **CHECK**, means must return true on success

Sorry, but that's simply wrong. "Check" is [imperative](https://www.eslbuzz.com/imperative-sentences-definition-examples/) and thus the function should return an integer value.

Predicative names would be `status_can_use_skill_mapzone` or `status_use_skill_mapzone_check_passed` for example.

Reply:
how about 
`status_check_skilluse_mapzone_validate?`
https://github.com/HerculesWS/Hercules/blob/da14478a8c0c616a6aa5481694c550143bc9b9f3/src/map/achievement.h#L226-L232

hahaha~

this is one thing I dont back down
https://github.com/HerculesWS/Hercules/blob/stable/src/map/achievement.c#L149-L164
see achievement.h, over there has check_complete, check_criteria, check_title in bool type function
https://github.com/HerculesWS/Hercules/blob/da14478a8c0c616a6aa5481694c550143bc9b9f3/src/map/achievement.h#L247-L251

Reply:
All the functions you posted are examples for how to not do it. There's no reason to imitate it.

> unless I see a PR that change those, then I also change mine

Well, I'm not in the mood to argue. Do as you like then... :disappointed: